doc/port-si/port-si-per-speed.md

```diff
@@ -0,0 +1,267 @@
+
+</br></br></br></br>
+
+
+# Feature Motivation
+
+
+Customers want to have complete flexibility and independence between switch vendors and optical cables.
+However, today, if our customers want to have our switches support some module of other vendors, they have to bear long waiting times for a FW release to come out. 
+So the goal of this feature is to allow them to take our switches and easily use them interchangeably with other vendors modules, and vice versa.
+Hence, a strategic decision has been made to transition the management of CMIS modules from FW to software SW.
+
+
+</br></br></br></br>
+
+
+
+# Design
+
+Currently, the configuration of SerDes SI is done entirely by FW without any involvement of SONiC. However, once we transition to independent mode, SONiC will take over the responsibility of configuring the SerDes. It will achieve this by sending the relevant port SI parameters to FW.</br>
+This flow mostly existed before, but was not put into practice because FW was responsible for configuring the SerDes. In the new mode, this flow will be expanded to support the new SI parameters sent to FW and to support the ability to send them based on the relevant lane speed. 
+Additionally, as part of working on this entire feature (outside the context of port SI per speed), the synchronization between the configuration of the ASIC and the configuration of the modules will be improved. In the new mode, once the ASIC configuration is complete, a signal will be sent from the FW to indicate that the module configuration can begin.
+ </br></br>
+ 
+
+# New SERDES parameters (data)
+
+## Application DB Enhancements
+
+6 new fields **ob_m2lp**, **ob_alev_out**, **obplev**, **obnlev**, **regn_bfm1p**, **regn_bfm1n** will be added to **PORT_TABLE** table:
+
+```
+ob_m2lp             = 1*8HEXDIG *( "," 1*8HEXDIG) ; list of hex values, one per lane              ; ratio between the central eye to the upper and lower eyes (for PAM4 only)
+ob_alev_out         = 1*8HEXDIG *( "," 1*8HEXDIG) ; list of hex values, one per lane              ; output common mode
+obplev              = 1*8HEXDIG *( "," 1*8HEXDIG) ; list of hex values, one per lane              ; output buffers input to Common mode PMOS side
+obnlev              = 1*8HEXDIG *( "," 1*8HEXDIG) ; list of hex values, one per lane              ; output buffers input to Common mode NMOS side
+regn_bfm1p          = 1*8HEXDIG *( "," 1*8HEXDIG) ; list of hex values, one per lane              ; voltage regulator to pre output buffer PMOS side
+regn_bfm1n          = 1*8HEXDIG *( "," 1*8HEXDIG) ; list of hex values, one per lane              ; voltage regulator to pre output buffer NMOS side
+```
+
+Here is the table to map the fields and SAI attributes:
+
+| Parameter     |              sai_port_attr_t                      |
+|:-------------:|:-------------------------------------------------:|
+| ob_m2lp	      |     SAI_PORT_SERDES_ATTR_TX_PAM4_RATIO            |
```

Comment:
Why do we have parameter names different in appl_db compared to how they are defined in SAI object. I don't think we should have them same terminology was our hardware/sdk fields. For e.g SAI_PORT_SERDES_ATTR_TX_PAM4_RATIO the corresponding field can be named tx_pam4_ratio instead of ob_m2lp